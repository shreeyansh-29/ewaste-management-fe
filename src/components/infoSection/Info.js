/* eslint-disable react/prop-types */
import React from "react";
import {Container} from "../../container/globalStyles";
import 'react-lazy-load-image-component/src/effects/blur.css';
import {
  InfoSec,
  InfoRow,
  InfoColumn,
  TextWrapper,
  TopLine,
  Heading,
  Subtitle,
  ImgWrapper,
  Img,
} from "./Info.elements";

const Info = ({
  lightBg,
  imgStart,
  lightTopLine,
  lightTextDesc,
  description,
  headline,
  topLine,
  lightText,
  img,
  alt,
  start,
}) => {
  return (
    <>
      <InfoSec lightBg={lightBg}>
        <Container>
          <InfoRow imgStart={imgStart}>
            <InfoColumn>
              <ImgWrapper start={start}>
                <Img src={img} alt={alt} effect="lazy" />
              </ImgWrapper>
            </InfoColumn>
            <InfoColumn>
              <TextWrapper>
                <TopLine lightTopLine={lightTopLine}>{topLine}</TopLine>
                <Heading lightText={lightText}>{headline}</Heading>
                <Subtitle lightTextDesc={lightTextDesc}>{description}</Subtitle>
              </TextWrapper>
            </InfoColumn>
          </InfoRow>
        </Container>
      </InfoSec>
    </>
  );
};

export default Info;
