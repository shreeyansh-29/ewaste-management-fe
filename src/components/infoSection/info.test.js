/* eslint-disable no-undef */
import React from "react";
import Info from "./Info";
import Enzyme, {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

Enzyme.configure({adapter: new Adapter()});

it("test infoSec", () => {
  expect.assertions(1);
  expect(toJson(shallow(<Info />))).toMatchSnapshot();
});
