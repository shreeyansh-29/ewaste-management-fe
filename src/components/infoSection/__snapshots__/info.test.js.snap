// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test infoSec 1`] = `
<Fragment>
  <styled.div>
    <styled.div>
      <styled.div>
        <styled.div>
          <styled.div>
            <styled.img />
          </styled.div>
        </styled.div>
        <styled.div>
          <styled.div>
            <styled.div />
            <styled.h1 />
            <styled.p />
          </styled.div>
        </styled.div>
      </styled.div>
    </styled.div>
  </styled.div>
</Fragment>
`;
